src/session_client.c

```diff
@@ -2452,10 +2452,12 @@ nc_recv_notif_thread(void *arg)
         if (msgtype == NC_MSG_NOTIF) {
             notif_clb(session, envp, op, user_data);
             if (!strcmp(op->schema->name, "notificationComplete") && !strcmp(op->schema->module->name, "nc-notifications")) {
-                lyd_free_tree(envp);
+                lyd_free_all(envp);
+                lyd_free_all(op);
                 break;
             }
-            lyd_free_tree(envp);
+            lyd_free_all(envp);
+            lyd_free_all(op);
```

Comment:
Found a memory leak here when using `netopeer2-cli` to test this. Not sure if this is the right way to free the memory here.

Reply:
Seems fine except I would expect (which is probably the problem here) these 2 trees connected.